core/src/main/java/org/apache/gravitino/storage/relational/mapper/provider/base/JobTemplateMetaBaseSQLProvider.java

```diff
@@ -0,0 +1,123 @@
+        + " #{jobTemplateMeta.jobTemplateContent}, #{jobTemplateMeta.auditInfo},"
+        + " #{jobTemplateMeta.currentVersion}, #{jobTemplateMeta.lastVersion},"
+        + " #{jobTemplateMeta.deletedAt})"
+        + " ON DUPLICATE KEY UPDATE"
+        + " job_template_name = #{jobTemplateMeta.jobTemplateName},"
+        + " metalake_id = #{jobTemplateMeta.metalakeId},"
+        + " job_template_comment = #{jobTemplateMeta.jobTemplateComment},"
+        + " job_template_content = #{jobTemplateMeta.jobTemplateContent},"
+        + " audit_info = #{jobTemplateMeta.auditInfo},"
+        + " current_version = #{jobTemplateMeta.currentVersion},"
+        + " last_version = #{jobTemplateMeta.lastVersion},"
+        + " deleted_at = #{jobTemplateMeta.deletedAt}";
+  }
+
+  public String listJobTemplatePOsByMetalake(@Param("metalakeName") String metalakeName) {
+    return "SELECT jtm.job_template_id AS jobTemplateId, jtm.job_template_name AS jobTemplateName,"
+        + " jtm.metalake_id AS metalakeId, jtm.job_template_comment AS jobTemplateComment,"
+        + " jtm.job_template_content AS jobTemplateContent, jtm.audit_info AS auditInfo,"
+        + " jtm.current_version AS currentVersion, jtm.last_version AS lastVersion,"
+        + " jtm.deleted_at AS deletedAt"
+        + " FROM "
+        + JobTemplateMetaMapper.TABLE_NAME
+        + " jtm JOIN "
+        + MetalakeMetaMapper.TABLE_NAME
+        + " mm ON jtm.metalake_id = mm.metalake_id"
+        + " WHERE mm.metalake_name = #{metalakeName} AND jtm.deleted_at = 0 AND mm.deleted_at = 0";
+  }
+
+  public String selectJobTemplatePOByMetalakeAndName(
+      @Param("metalakeName") String metalakeName,
+      @Param("jobTemplateName") String jobTemplateName) {
+    return "SELECT jtm.job_template_id AS jobTemplateId, jtm.job_template_name AS jobTemplateName,"
+        + " jtm.metalake_id AS metalakeId, jtm.job_template_comment AS jobTemplateComment,"
+        + " jtm.job_template_content AS jobTemplateContent, jtm.audit_info AS auditInfo,"
+        + " jtm.current_version AS currentVersion, jtm.last_version AS lastVersion,"
+        + " jtm.deleted_at AS deletedAt"
+        + " FROM "
+        + JobTemplateMetaMapper.TABLE_NAME
+        + " jtm JOIN "
+        + MetalakeMetaMapper.TABLE_NAME
+        + " mm ON jtm.metalake_id = mm.metalake_id"
+        + " WHERE mm.metalake_name = #{metalakeName} AND jtm.job_template_name = #{jobTemplateName}"
+        + " AND jtm.deleted_at = 0 AND mm.deleted_at = 0";
+  }
+
+  public String softDeleteJobTemplateMetaByMetalakeAndName(
+      @Param("metalakeName") String metalakeName,
+      @Param("jobTemplateName") String jobTemplateName) {
+    return "UPDATE "
+        + JobTemplateMetaMapper.TABLE_NAME
+        + " SET deleted_at = (UNIX_TIMESTAMP() * 1000.0)"
+        + " + EXTRACT(MICROSECOND FROM CURRENT_TIMESTAMP(3)) / 1000.0"
+        + " WHERE job_template_name = #{jobTemplateName} AND metalake_id ="
+        + " (SELECT metalake_id FROM "
```

Comment:
ditto, should use `IN` as follows.

Reply:
If it returns multiple job template id, then there should be a big logic problem, using `IN` just hide the problem.

Reply:
This should not be happened, right? Metalake name should be globally unique.

Reply:
I see and I made a mistake here.